lib/tls.js

```diff
@@ -53,7 +53,8 @@ exports.convertNPNProtocols = function(protocols, out) {
   }
   // If it's already a Buffer - store it
   if (protocols instanceof Buffer) {
-    out.NPNProtocols = protocols;
+    // copy new buffer not to be modified by user
+    out.NPNProtocols = Buffer.from(protocols);
```

Comment:
This is inefficient (in a minor way) if `protocols` was originally an array.  In that case two buffers are created, here and in `convertProtocols()`.